src/saveToCase.js

```diff
@@ -632,7 +632,7 @@ define([
                 if (caseIdRegex.test(value.ref)) {
                     mug.p.case_id = value.value;
                     mug.form.dropSetValues(function(inner) {
-                        return value.value === inner.value;
+                        return value.ref === inner.ref;
```

Comment:
I'm going to punt on that for this PR, until I can think more about it.

First thoughts:

1) we don't guarantee ordering of setvalues (I believe they are in mug order, and then any extra ones get added to the end), which could be confusing if there's random setvalues. 
2) save_to_case is a (somewhat) supported question with a known case_id there should only be one in the form, anything else is a bug (although would be great to have a javascript `soft_assert` for this).